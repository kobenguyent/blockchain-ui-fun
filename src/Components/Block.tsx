import {Button, Card, CardGroup, Container, ListGroup} from "react-bootstrap";

export const Block = ({ block, setSelectedBlockIndex, txCreation, setTxCreation, showPendingTxList }:any) => {
    if (txCreation === false) {
        setTxCreation(false)

        if (!showPendingTxList) {
            return (
                <Container>
                    <h1>Blocks on chain</h1>
                    Each card represents a block on the chain. Click on a block to see the transactions stored inside.
                    <CardGroup>
                        {block.map((item:any, i:any) => {
                            return (
                                <Card className="g-2" border="primary" key={i} onClick={() => setSelectedBlockIndex(i)}>
                                    <Card.Body>
                                        <Card.Title>Block #{i + 1}</Card.Title>
                                        <Card.Text>
                                            <ListGroup className="list-group-flush">
                                                <ListGroup.Item>Hash
                                                    <div className="text-truncate" style={{"color": '#' + item.hash.substring(0,6) }}>{item.hash}</div>
                                                </ListGroup.Item>
                                                <ListGroup.Item>Previous hash
                                                    <div className="text-truncate" style={{"color": '#' + item.prevHash.substring(0,6) }}>{item.prevHash}</div>
                                                </ListGroup.Item>
                                                <ListGroup.Item> Nonce
                                                    <div className="text-truncate" style={{"color": 'grey' }}>{item.nonce}</div>
                                                </ListGroup.Item>
                                                <ListGroup.Item>Timestamp
                                                    <div className="text-truncate" style={{"color": 'grey' }}>{item.timestamp}</div>
                                                </ListGroup.Item>
                                            </ListGroup>
                                            <Button variant="primary">Show Transactions</Button>
                                        </Card.Text>
                                    </Card.Body>
                                </Card>
                            );
                        })}
                    </CardGroup>
                </Container>
            )
        }

    }

    return (
        <></>
    )

}
