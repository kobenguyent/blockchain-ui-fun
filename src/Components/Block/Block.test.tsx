import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import {Block} from "./Block.tsx";

describe('Block component', async () => {
    it.only('Should render the block component correctly', async () => {
        const setSelectedBlockIndex = (index:any) => {
            return index
        }
        const blockchain = [
            {
                "timestamp": 1682410957432,
                "transactions": [],
                "prevHash": "0",
                "hash": "8a48e33caabc3a1b04a3aee52d9614b1e838fa5e99eac79197ea7fc92c46df99",
                "nonce": 0
            }]
        // Setup
        render(<Block block={blockchain} setSelectedBlockIndex={setSelectedBlockIndex}/>);

        const h1 = await screen.getByText('Blocks on chain')

        expect(h1).toMatchSnapshot()

    });
});
