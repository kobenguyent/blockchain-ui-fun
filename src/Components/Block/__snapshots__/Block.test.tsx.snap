// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Block component > Should render the block component correctly 1`] = `
<h1>
  Blocks on chain
</h1>
`;
